js/src/admin/components/BasicsPage.js

```diff
@@ -40,157 +20,109 @@ export default class BasicsPage extends Page {
-                    <Select options={options} value={this.values[`slug_driver_${model}`]()} onchange={this.values[`slug_driver_${model}`]}></Select>
-                  </FieldSet>
-                );
-              }
-            })}
-
-            {Button.component(
-              {
-                type: 'submit',
-                className: 'Button Button--primary',
-                loading: this.loading,
-                disabled: !this.changed(),
-              },
-              app.translator.trans('core.admin.basics.submit_button')
-            )}
-          </form>
+  content() {
+    return [
+      <div className="Form">
+        {this.buildSettingComponent({
+          type: 'text',
+          setting: 'forum_title',
+          label: app.translator.trans('core.admin.basics.forum_title_heading'),
+        })}
+        {this.buildSettingComponent({
+          type: 'text',
+          setting: 'forum_description',
+          label: app.translator.trans('core.admin.basics.forum_description_heading'),
+          help: app.translator.trans('core.admin.basics.forum_description_text'),
+        })}
+
+        {Object.keys(this.localeOptions).length > 1
+          ? [
+              this.buildSettingComponent({
+                type: 'select',
+                setting: 'default_locale',
+                options: this.localeOptions,
+                label: app.translator.trans('core.admin.basics.default_language_heading'),
+              }),
+              this.buildSettingComponent({
+                type: 'switch',
+                setting: 'show_language_selector',
+                label: app.translator.trans('core.admin.basics.show_language_selector_label'),
+              }),
+            ]
+          : ''}
+
+        {FieldSet.component(
+          {
+            label: app.translator.trans('core.admin.basics.home_page_heading'),
+            className: 'BasicsPage-homePage Form-group',
+          },
+          [
+            <div className="helpText">{app.translator.trans('core.admin.basics.home_page_text')}</div>,
+            this.homePageItems()
+              .toArray()
+              .map(({ path, label }) => (
+                <label className="checkbox">
+                  <input type="radio" name="homePage" value={path} bidi={this.setting('default_route')} />
+                  {label}
+                </label>
+              )),
+          ]
+        )}
+
+        <div className="BasicsPage-welcomeBanner-input">
+          {this.buildSettingComponent({
+            type: 'text',
+            setting: 'welcome_title',
+            label: app.translator.trans('core.admin.basics.welcome_banner_heading'),
+            className: 'BasicsPage-welcomeBanner',
+            help: app.translator.trans('core.admin.basics.welcome_banner_text'),
+          })}
+          <textarea className="FormControl" bidi={this.setting('welcome_message')} />
```

Comment:
I would prefer to directly use JSX for this one since it's a custom setting. The advantage to directly using JSX is that the HTML structure remains consistent in that `.FormGroup` will always be the direct child of `.Form` and will contain the setting inputs, whereas currently it doesn't make much sense.

```jsx
<div className="Form-group">
  <label>{app.translator.trans('core.admin.basics.welcome_banner_heading')}</label>
  <div className="helpText">{app.translator.trans('core.admin.basics.welcome_banner_text')}</div>
  <input type="text" bidi={this.setting('welcome_title')} />
  <textarea className="FormControl" bidi={this.setting('welcome_message')} />
</div>
```